modules/earthengine-layers/src/earth-engine-layer.js

```diff
@@ -25,10 +32,21 @@ export default class EarthEngineLayer extends CompositeLayer {
   }
 
   updateState({props, oldProps, changeFlags}) {
+    this._updateToken(props, oldProps, changeFlags);
```

Comment:
Does it matter that there's no `await` here?

Reply:
No, since I assume _updateToken is "awaiting" and doing the needful updates, you are just triggering the operation here.

It may look more cleart to make updateState async and await, but I don't think anyone is "awaiting" updateState.

Reply:
Well my understanding of async code isn't yet bulletproof... Without an `await` on `this._updateToken`, that triggers the token update, but the event loop could move to `this._updateEEObject` before `this._updateToken` finishes?

Calling any `ee` method before `ee.Initialize` raises an exception, e.g. in Node:
```js
var ee = require('@google/earthengine');
var json_str = '{"type": "Invocation", "arguments": {"id": "LANDSAT/LT05/C01/T1_SR/LT05_034033_20000913"}, "functionName": "Image.load"}'
ee.Deserializer.fromJSON(json_str)
// Error: Request is missing required authentication credential
```

So without an `await` on `this._updateToken` is it possible that `ee.Deserializer.fromJSON` within `_updateEEObject` could be called before `ee.Initialize` finishes?

Reply:
Yes I see what you mean. You should be able to do an await here. It will force every update to be async generating some extra render but that is OK.